File: 4-strings/2-string-basics/main.cpp
// String Basics - C++ Example

#include <iostream>
#include <string>

int main() {
    std::string text = "hello";

    std::cout << "Length: " << text.length() << std::endl;
    std::cout << "First character: " << text[0] << std::endl;
    std::string upper_text;
    for (char c : text) upper_text += toupper(c);
    std::cout << "Uppercase: " << upper_text << std::endl;
    std::cout << "Contains 'll': " << (text.find("ll") != std::string::npos ? "true" : "false") << std::endl;

    return 0;
}

File: 4-strings/2-string-basics/main.java
// String Basics - Java Example

public class StringBasics {

    public static void main(String[] args) {
        String text = "hello";

        System.out.println("Length: " + text.length());
        System.out.println("First character: " + text.charAt(0));
        System.out.println("Uppercase: " + text.toUpperCase());
        System.out.println("Contains 'll': " + text.contains("ll"));
    }
}
